sphinx/ext/autodoc.py

```diff
@@ -284,7 +284,7 @@ def format_annotation(annotation):
                 return '%s[%s]' % (qualified_name, param_str)
         elif hasattr(typing, 'CallableMeta') and \
                 isinstance(annotation, typing.CallableMeta) and \
-                hasattr(annotation, '__args__') and \
+                getattr(annotation, '__args__', None) is not None and \
```

Comment:
Very clever patch.  Could you add a comment that indicates we let bare typing.Callable fall through here?